setup.py

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+from setuptools import (
+    setup,
+    find_packages,
+)
+
+extras_require = {
+    'test': [
+        "pytest==5.4.1",
+        "pytest-xdist",
+        "tox==3.14.6",
+    ],
+    'lint': [
+        "flake8==3.7.9",
+        "isort>=4.2.15,<5",
+        "mypy==0.770",
+        "pydocstyle>=3.0.0,<4",
+    ],
+    'doc': [
+        "Sphinx>=1.6.5,<2",
+        "sphinx_rtd_theme>=0.1.9",
+        "towncrier>=19.2.0, <20",
+    ],
+    'dev': [
+        "bumpversion>=0.5.3,<1",
+        "pytest-watch>=4.1.0,<5",
+        "wheel",
+        "twine",
+        "ipython",
+    ],
+}
+
+extras_require['dev'] = (
+    extras_require['dev'] +  # noqa: W504
+    extras_require['test'] +  # noqa: W504
+    extras_require['lint'] +  # noqa: W504
+    extras_require['doc']
+)
+
+
+with open('./README.md') as readme:
+    long_description = readme.read()
+
+
+setup(
+    name='<PYPI_NAME>',
```

Comment:
It'd probably be good to stub these out with something a bit more appropriate to the project.

Reply:
> If you want a short and quick way to get everything `flake8` compliant you can use [`black`](https://github.com/psf/black) to automatically reformat the codebase. It's probably the shortest path from where we are now to a PEP8 compliant codebase, although it will change a **LOT** of code.

Interesting point, I will keep in mind. However, for now, I would prefer to keep the current settings and let #81 fix flake8 issues. 

Besides, I am not sure if it is able to fix them all. For example, there were few issues with long strings (length >100 characters) and my linter couldn't fix it automatically. I guess it is because the fix requires splitting the string and merging them, which implies some semantic changes.